src/callbacks.cpp

```diff
@@ -0,0 +1,231 @@
+                 "WaypointNode::%s: Received position from "
+                 "'%s': lat: %f, lon: %f, height: %f",
+                 __func__, msg.sender_id.c_str(), msg.latitude_deg,
+                 msg.longitude_deg, msg.relative_altitude_m);
+
+    // Check timestamp
+    if (timestamp_now - rclcpp::Time(msg.time_stamp) >
+        rclcpp::Duration(std::chrono::duration<int64_t, std::milli>(
+            max_position_msg_time_difference_ms))) {
+        // Message is too old, react based on the current 'active' state
+
+        // Reset pos as we can no longer reliabely know where we are
+        pos = Position();
+
+        if (this->get_active()) {
+            // Abort job if currently active
+
+            RCLCPP_FATAL(get_logger(),
+                         "WaypointNode::%s: Received too old "
+                         "timestamp in position message: %s. Aborting Job.",
+                         __func__, msg.sender_id.c_str());
+            this->job_finished("WaypointNode::" + (std::string) __func__ +
+                               ": A too old timestamp was received in a "
+                               "position message while "
+                               "being active");
+            reset_node();
+        } else {
+            // Warn but still store values
+
+            RCLCPP_WARN(get_logger(),
+                        "WaypointNode::%s: Received too old "
+                        "timestamp in position message: %s",
+                        __func__, msg.sender_id.c_str());
+        }
+    }
+
+    // Store values
+    pos.values_set = true;
+    pos.coordinate_lat = msg.latitude_deg;
+    pos.coordinate_lon = msg.longitude_deg;
+    pos.height_cm = msg.relative_altitude_m * 100.0;
+}
+
+/**
+ * @brief Callback function for receiving mission progress messages.
+ *
+ * This function is called when a mission progress message is received. It
+ * updates the mission progress of the WaypointNode based on the received
+ * message.
+ *
+ * @param msg The mission progress message received.
+ */
+void WaypointNode::callback_mission_progress(
+    const interfaces::msg::MissionProgress &msg) {
+    // Store current timestamp for later
+    rclcpp::Time timestamp_now = this->now();
+
+    // Ignore message if node is not active
+    if (!this->get_active()) return;
+
+    RCLCPP_DEBUG(this->get_logger(),
+                 "WaypointNode::%s: Received mission "
+                 "progress from '%s': progress: %f / 1.0",
+                 __func__, msg.sender_id.c_str(), msg.progress);
+
+    // Check timestamp
+    if (timestamp_now - rclcpp::Time(msg.time_stamp) >
+        rclcpp::Duration(std::chrono::duration<int64_t, std::milli>(
+            max_progress_msg_time_difference_ms))) {
+        // Warn
+        RCLCPP_WARN(
+            get_logger(),
+            "WaypointNode::%s: Received too "
+            "old timestamp in progress message: %s. Using value regardless.",
+            __func__, msg.sender_id.c_str());
+    }
+
```

Comment:
Maybe check the sender id as well.

Reply:
https://github.com/THI-Drone/waypoint_package/pull/3#discussion_r1598613924